[addons][pvr] remove more deprecated parts and make PVR tables in own parts

AlwinEsch opened the pull request:
## Description

Commit 1:
---------------------
**[pvr] remove deprecated `inputstreamaddon` property**

This done by another named `inputstreamclass`.

Commit 2:
------------------------
**[pvr] make props and functions to/from Kodi in own parts**

This is to prevent problems by add of something where has needed before
a complete API increase (if was something on props or callbacks).

Now have every part a own created memory where can be changed without
attack of followed parts.

Commit 3:
------------------------

**[addons][pvr] remove deprecated kodi_vfs_types.h**

The two structures currently used in it are moved to libXBMC_addon.h
and kodi_vfs_types.h deleted.

The rest of them was used years before for the old VFS addon way.

Commit 4:
------------------------

**[addons][pvr] remove deprecated interface parts**

This remove some parts where since long time no more used or
present on other place.

Commit 5:
------------------------
**[addons][pvr] move xbmc_epg_types.h to xbmc_pvr_types.h**

This is another change to clean up the old interface where the PVR
related EPG data is moved to it and xbmc_epg_types.h deleted.

Also is the EPG_EVENT_CONTENTMASK added to a enum where match more his type.

Commit 6:
------------------------
**[addons][pvr] correct CPVRClient::GetDriveSpace call**

This change is to make it equal to all other calls and use given
"const AddonInstance* addon" value.

Commit 7:
------------------------
**[addons][pvr] use STREAM_MAX_PROPERTY_COUNT from inputstream also for pvr**

Thought to use the same on pvr and inputstream to confirm both places have the
same size.

This can prevent some update problems, for PVR this not related to his
sources, as they not use it. Only used for PVR inside Kodi.

Commit 8:
------------------------
**[addons][pvr] remove stream properties and do over inputstream**

Before was them in inputstream and the same as copy in PVR, this now remove it
there to use also them from inputstream. This to make in case of change more
sure to have equal and to have it cleaner.

Commit 9:
------------------------
**increase PVR API version to 6.4.0**

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
To reduce further the size of https://github.com/xbmc/xbmc/pull/16485 and take unrelated minor things from there.

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons][pvr] remove deprecated interface parts

This remove some parts where since long time no more used or
present on other place.

AlwinEsch committed:
[addons][pvr] move xbmc_epg_types.h to xbmc_pvr_types.h

This is another change to clean up the old interface where the PVR
related EPG data is moved to it and xbmc_epg_types.h deleted.

AlwinEsch committed:
[addons][pvr] correct CPVRClient::GetDriveSpace call

This change is to make it equal to all other calls and use given
"const AddonInstance* addon" value.

AlwinEsch committed:
[addons][pvr] increase PVR API version to 6.4.0

phunkyfish commented:
Just some high level notes, generally it looks ok.

**Commit 1: [pvr] remove deprecated inputstreamaddon property**

Agree this needs should be removed but will probably need a forum post to let users know the change is happening. IPTV Simple mainly as many users construct their own M3U files and all that will break when this is removed. Could also leave it in the addons and have them convert it to the right version. At least then the extra code is only in 

**Commit 5: [addons][pvr] move xbmc_epg_types.h to xbmc_pvr_types.h**

There must have been a reason to these to be separate in the first place. Do you know it?

**Commit 6:[inputstream][pvr] rename define STREAM_PROPERTY_INPUTSTREAMCLASS**

I think for readability I would either leave this as is or change it completely including changing `inputstreamclass` to `inputstreamname`. `inpustreamclass` was only introduced in alpha so maybe not sure a big deal to change it.

Personally I would probably leave it as is. It's an identifier to allow kodi to select the right class to load for an inputstream.

AlwinEsch commented:
Thanks, for review!

With Commit 2, the idea was to use EPG independently of the PVR. At some point in 2008-2009 I created it this way :smirk:.

Because of commit 1, it would make sense to post a forum. 
Then leave it in the code, only then remove it from the PVR.

About **`STREAM_PROPERTY_INPUTSTREAMCLASS`** comes from here https://github.com/xbmc/xbmc/pull/16485#discussion_r366324944, was not sure how much this used by Python addons, but when not on Leia, it could be renamed there too.

Using its name as a `CLASS` is a little confusing since it actually refers to an addon or ffmpeg in Kodi.  As a `*_CLASS`, I expect a C++ class in Kodi.

phunkyfish commented:
> About **`STREAM_PROPERTY_INPUTSTREAMCLASS`** comes from here [#16485 (comment)](https://github.com/xbmc/xbmc/pull/16485#discussion_r366324944), was not sure how much this used by Python addons, but when not on Leia, it could be renamed there too.
> 
> Using its name as a `CLASS` is a little confusing since it actually refers to an addon or ffmpeg in Kodi. As a `*_CLASS`, I expect a C++ class in Kodi.

It's not in Leia and was only added to Matrix while in alpha. Similar to `inpustreamaddon` you could leave `inputstreamclass` there for a few months and then deprecate it. A pity I did not introduce `inputstream` name then instead!

AlwinEsch committed:
[pvr] remove deprecated "inputstreamaddon" property

This done by another named "inputstreamclass".

AlwinEsch commented:
Have leaved now by `inputstreamclass` and add on the rework a detailed documentation about to know what it do.

AlwinEsch committed:
[pvr] make props and functions to/from Kodi in own parts

This is to prevent problems by add of something where has needed before
a complete API increase (if was something on props or callbacks).

Now have every part a own created memory where can be changed without
attack of followed parts.

AlwinEsch commented:
jenkins build this please

phunkyfish commented:
Maybe add the increase in stream properties to 30 too?

AlwinEsch commented:
@phunkyfish have added a new commit about, also OK to bring now in.

phunkyfish reviewed (approved):
Looks good, once Jenkins is happy.

phunkyfish commented:
Jenkins build this please

phunkyfish commented:
jenkins build this please

AlwinEsch commented:
Notice from a stuipd man like me, have added the `#include <time.h>` to wrong header.

AlwinEsch committed:
[addons][pvr] remove deprecated kodi_vfs_types.h

The two structures currently used in it are moved to libXBMC_addon.h
and kodi_vfs_types.h deleted.

AlwinEsch commented:
Jenkins now happy :smile:

phunkyfish commented:
Nice 😉

phunkyfish commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h:
```diff
@@ -64,10 +41,20 @@ struct DemuxPacket;
-#define PVR_STREAM_MAX_PROPERTIES     20
 #define PVR_STREAM_PROPERTY_STREAMURL "streamurl" /*!< @brief the URL of the stream that should be played. */
-#define PVR_STREAM_PROPERTY_INPUTSTREAMADDON  "inputstreamaddon" /*!< @brief the name of the inputstream add-on that should be used by Kodi to play the stream denoted by PVR_STREAM_PROPERTY_STREAMURL. Leave blank to use Kodi's built-in playing capabilities. */
 #define PVR_STREAM_PROPERTY_INPUTSTREAMCLASS  "inputstreamclass" /*!< @brief the name of the inputstream add-on that should be used by Kodi to play the stream denoted by PVR_STREAM_PROPERTY_STREAMURL. Leave blank to use Kodi's built-in playing capabilities or to allow ffmpeg to handle directly set to `inputstream.ffmpeg`. */
```
Just a reminder to add:
```
#define PVR_STREAM_PROPERTY_INPUTSTREAM  "inputstream" /*!< @brief the name of the inputstream that should be used by Kodi to play the stream denoted by PVR_STREAM_PROPERTY_STREAMURL. Leave blank to use Kodi's built-in playing capabilities or to allow ffmpeg to handle directly set to `inputstream.ffmpeg`. */

AlwinEsch replied:
Thanks, I look to change.

AlwinEsch replied:
Thanks, I look to change.

phunkyfish replied:
You can also delete PVR_STREAM_PROPERTY_INPUTSTREAMCLASS

phunkyfish reviewed (commented):
That's it ;)

AlwinEsch committed:
[addons][pvr] use STREAM_MAX_PROPERTY_COUNT from inputstream also for pvr

Thought to use the same on pvr and inputstream to confirm both places have the
same size.

This can prevent some update problems, for PVR this not related to his
sources, as they not use it. Only used for PVR inside Kodi.

phunkyfish commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h:
```diff
@@ -64,15 +47,11 @@ struct DemuxPacket;
+// other general inputstream properties.
 #define PVR_STREAM_PROPERTY_STREAMURL "streamurl" /*!< @brief the URL of the stream that should be played. */
-#define PVR_STREAM_PROPERTY_INPUTSTREAMADDON  "inputstreamaddon" /*!< @brief the name of the inputstream add-on that should be used by Kodi to play the stream denoted by PVR_STREAM_PROPERTY_STREAMURL. Leave blank to use Kodi's built-in playing capabilities. */
-#define PVR_STREAM_PROPERTY_INPUTSTREAMCLASS  "inputstreamclass" /*!< @brief the name of the inputstream add-on that should be used by Kodi to play the stream denoted by PVR_STREAM_PROPERTY_STREAMURL. Leave blank to use Kodi's built-in playing capabilities or to allow ffmpeg to handle directly set to `inputstream.ffmpeg`. */
```
Would it be better to define this as:` #define PVR_STREAM_PROPERTY_INPUTSTREAM  STREAM_PROPERTY_INPUTSTREAM `

AlwinEsch replied:
Makes sene, I change to them.

phunkyfish commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h:
```diff
@@ -64,15 +47,11 @@ struct DemuxPacket;
-#define PVR_STREAM_PROPERTY_INPUTSTREAMADDON  "inputstreamaddon" /*!< @brief the name of the inputstream add-on that should be used by Kodi to play the stream denoted by PVR_STREAM_PROPERTY_STREAMURL. Leave blank to use Kodi's built-in playing capabilities. */
-#define PVR_STREAM_PROPERTY_INPUTSTREAMCLASS  "inputstreamclass" /*!< @brief the name of the inputstream add-on that should be used by Kodi to play the stream denoted by PVR_STREAM_PROPERTY_STREAMURL. Leave blank to use Kodi's built-in playing capabilities or to allow ffmpeg to handle directly set to `inputstream.ffmpeg`. */
 #define PVR_STREAM_PROPERTY_MIMETYPE "mimetype" /*!< @brief the MIME type of the stream that should be played. */
-#define PVR_STREAM_PROPERTY_ISREALTIMESTREAM "isrealtimestream" /*!< @brief "true" to denote that the stream that should be played is a realtime stream. Any other value indicates that this is no realtime stream.*/
```
Would it be better to define this as:` #define PVR_STREAM_PROPERTY_ISREALTIMESTREAM  STREAM_PROPERTY_ISREALTIMESTREAM `

phunkyfish commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h:
```diff
@@ -64,15 +47,11 @@ struct DemuxPacket;
 #define PVR_STREAM_PROPERTY_MIMETYPE "mimetype" /*!< @brief the MIME type of the stream that should be played. */
-#define PVR_STREAM_PROPERTY_ISREALTIMESTREAM "isrealtimestream" /*!< @brief "true" to denote that the stream that should be played is a realtime stream. Any other value indicates that this is no realtime stream.*/
 #define PVR_STREAM_PROPERTY_EPGPLAYBACKASLIVE "epgplaybackaslive" /*!< @brief "true" to denote that if the stream is from an EPG tag that it should be played is a live stream. Otherwise if it's a EPG tag it will play as normal video.*/
-#define PVR_STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG  "inputstream.ffmpeg" /*!< @brief special value for PVR_STREAM_PROPERTY_INPUTSTREAMCLASS to use ffmpeg to directly play a stream URL. */
```
Would it be better to define this as:` #define PVR_STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG  STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG `

phunkyfish reviewed (changes requested):
This keeps the defines in one place and the API clean for PVR. Otherwise dev won't know where to look for the values.

AlwinEsch committed:
[addons][pvr] change stream properties and do over inputstream

Before was them in inputstream and the same as copy in PVR, this now remove it
there to use also the defines from inputstream.
This to make in case of change more sure to have equal and to have it cleaner.

Further is the documentation a bit changes to match the rework part.

AlwinEsch commented:
Have it changed and also now on correct Doxygen style, where then bring like this:

![Bildschirmfoto vom 2020-04-19 20-50-59](https://user-images.githubusercontent.com/6879739/79697961-f3287e80-8285-11ea-914b-173058a38fd7.png)

phunkyfish commented:
That looks really good!

phunkyfish reviewed (approved):
👍

Rechi commented on xbmc/pvr/addons/PVRClient.cpp:
```diff
@@ -546,7 +560,7 @@ PVR_ERROR CPVRClient::GetDriveSpace(long long& iTotal, long long& iUsed)
   return DoAddonCall(__FUNCTION__, [this, &iTotal, &iUsed](const AddonInstance* addon) {
```
```
xbmc/pvr/addons/PVRClient.cpp:560:37: warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
  return DoAddonCall(__FUNCTION__, [this, &iTotal, &iUsed](const AddonInstance* addon) {
                                    ^~~~~
1 warning generated.
```

AlwinEsch replied:
Was it only be there or also on other places, e.g. `GetChannelGroups` ...?
Has you becomed them from Jenkins or from another tester, if yes have you a link to it?

Thanks for report about.

Rechi replied:
This is the only place.
https://jenkins.kodi.tv/job/Android-ARM/17742/clang/new/

AlwinEsch replied:
Have looked in jenkins that a good feature to see:
- GNU C Compiler (gcc) Warnings
- Clang (LLVM based) Warnings

Rechi replied:
This existed since August 2017, also did fail builds if new warnings got intoduced. So one could have a look at the code and fix it.
This was removed in March, for whatever reason.

AlwinEsch committed:
[addons][pvr] remove not used lambda for "this" on GetDriveSpace call

Rechi commented:
jenkins build this please

phunkyfish commented on xbmc/addons/interfaces/Addon/AddonCallbacksAddon.cpp:
```diff
@@ -102,21 +101,21 @@ void CAddonCallbacksAddon::AddOnLog(void *addonData, const addon_log_t addonLogL
+        xbmcLogLevel = LOGFATAL;
       default:
-        xbmcLogLevel = LOGDEBUG;
+        xbmcLogLevel = LOGNONE;
```
Maybe add a comment the line before saying:
`// If the log level is invalid then don't log anything`

phunkyfish replied:
Then in the future we know what this log level is for!

AlwinEsch committed:
[addons][pvr] remove obsolete LOG types and sync with Kodi side

The LOG_NOTICE is LOG_INFO. Also are the LOG_WARNING and LOG_FATAL added.

Further is the namespace ADDONS removed from them, to allow combine with
new interface on next requests.

AlwinEsch commented:
jenkins build this please

AlwinEsch commented:
jenkins build this please

AlwinEsch commented:
jenkins build this please
